doc/whats_new/v1.4.rst

```diff
@@ -35,6 +35,11 @@ random sampling procedures.
       solvers (when fit on the same data again). The amount of change depends on the
       specified `tol`, for small values you will get more precise results.
 
+- |Efficiency| :class:`preprocessing.QuantileTransformer` now uses `resample` function to
+  perform subsampling and remove the for-loop while computing quantiles. For dense matrices,
+  it will not get the same subsample examples with the same `random_state` as before.
```

Comment:
We don't need to mention the implementation details but just that the samples used to compute quantiles will be different leading to a change in when calling `transform`.

Reply:
done